compile-db-gen.py

```diff
@@ -125,33 +138,33 @@ def genlineobjs(fname):
                 pid, cid = m.group(1, 2)
                 if cid is not None:
                     parent[cid] = pid
-                else:
-                    unfinished_fork = True
-                continue
+                    continue
+                # otherwise, it's unfinished fork(), search the resumed one
+                flinum = fill_cache(fd, flinum, cache_lines)
+                for [linum, line] in cache_lines:
```

Comment:
What if `cache_lines` does not contain the line we are looking for? It seems that we will silently miss the entry in `parent`. It will probably lead to problems when we need this information in `parent` at later stages.

To be honest, I don't understand why `cache_lines` is needed. We can process the trace line by line and sooner or later we will find the matching resumed syscall. Sort of like `fork_re` and `fork_resumed_re` used to work before these changes. Did you experience any issues with this approach?

`cache_lines` seems to only add complexity, both in terms of code to maintan and algorithmic complexity. It introduces back some of O(N<sup>2</sup>) complexity I was fighting in #5.

Reply:
You'r right, it will be dropped silently. 
The `cache_lines` is try to deal follow lines, in which the `exeve` between the `fork unfinished` and `fork resumed`, the parser run to the execve, but can not get its parent.
>4384  vfork( <unfinished ...>
4385  execve("/usr/libexec/gcc/x86_64-redhat-linux/4.8.5/cc1", ["/usr/libexec/gcc/x86_64-redhat-linux/4.8.5/cc1", "-quiet", "-I", "../include", "-I", "../lib", "-I", "../lib", "-MD", ".libs/libcurl_la-altsvc.d", "-MF", ".deps/libcurl_la-altsvc.Tpo", "-MP", "-MT", "libcurl_la-altsvc.lo", "-D", "HAVE_CONFIG_H", "-D", "BUILDING_LIBCURL", "-D", "CURL_HIDDEN_SYMBOLS", "-D", "PIC", "altsvc.c", "-quiet", "-dumpbase", "altsvc.c", "-mtune=generic", "-march=x86-64", "-auxbase-strip", ".libs/libcurl_la-altsvc.o", "-O2", "-Werror=implicit-function-declaration", "-Wno-system-headers", "-fvisibility=hidden", "-fPIC", "-o", "/tmp/ccbyVQNG.s"], 0x1e0edf0 /* 45 vars */ <unfinished ...>
4384  <... vfork resumed>)              = 4385

Reply:
> the parser run to the `execve` line (line 2), but can not get its parent pid, and the parent pid is avaliable on line 3

As far as I understand, this should not be a problem, because there are two "passes". On the first pass strace output is processed into a list of "events" (`obj_list`) and `parent` is filled up. The child-parent relations from `parent` are only needed on the second pass, when `obj_list` is processed further. By that time we will already have a complete picture in `parent`.